### PR TITLE
Typing `tuner.auto_gpu_select`

### PR DESCRIPTION
**Pull request by jjenniferdai**
## What does this PR do?

Part of #5023, add typing for `tuner.auto_gpu_select.py`

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/tuner/auto_gpu_select.py [hunk 1]
```diff
@@ -30,14 +32,14 @@ def pick_multiple_gpus(nb):
 
     nb = torch.cuda.device_count() if nb == -1 else nb
 
-    picked = []
+    picked = []  # type: List[int]
```
Comment (ananthsub) [inline pytorch_lightning/tuner/auto_gpu_select.py:35]: does it work to do this?
```suggestion
    picked: List[int] = []
```

Comment (jjenniferdai) [reply pytorch_lightning/tuner/auto_gpu_select.py:35]: yea I think this is [better](https://mypy.readthedocs.io/en/stable/getting_started.html#local-type-inference) i'll use this here. (just saw some comment syntax `# type: ignore` in the codebase)

File: pyproject.toml [hunk 2]
```diff
@@ -76,5 +76,6 @@ module = [
     "pytorch_lightning.utilities.memory",
     "pytorch_lightning.utilities.parsing",
     "pytorch_lightning.utilities.xla_device",
+    "pytorch_lightning.tuner.auto_gpu_select",
```
Comment (ananthsub) [inline pyproject.toml:79]: let's keep this list alphabetized

**Review by ananthsub — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by SkafteNicki — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Pull request merged**